### DISCUSSION TITLE
Why does SerialWriteLine not work in this instance?

### DISCUSSION THREAD

**Discussion by hengweizhu** [bonsai-rx/bonsai#2078] [Q&A]
I am capturing images from a Chameleon3 u3 13y3m-cs camera (PointGrey/FlyCapture). I want to send a string to my Arduino (e.g. the letter 'x') for the Arduino to then output a TTL pulse to signify when it has captured the image. For some reason, the Arduino does not output the TTL when I attach it to any part of the FlyCapture chain. It does do it when it is not in that chain of nodes. Can anyone help me with this please? I've attach an image of the workflow.

![arenacam_workflow_pt1](https://github.com/user-attachments/assets/f549780b-4d7d-4ca9-9dbb-e2c84d984318)
![image](https://github.com/user-attachments/assets/0d5d433c-16f8-4085-a1e4-8c8cfe3c95b6)

**Comment by bruno-f-cruz** [comment]
Hi,
Include the workflow as asked in #864 
Thanks!

**Comment by hengweizhu** [comment]
[arena_workflow.zip](https://github.com/user-attachments/files/18113417/arena_workflow.zip)
Hi Bruno,

I've attached the workflow in the zip file as instructed. It also contains the simple arduino script that controls COM4 which is the SerialWriteLine part attached to the FlyCapture that I am having issues with. Thanks!

**Reply by bruno-f-cruz** [reply]
Can you try moving the ´CreateSerialPort´ to the top of the workflow? I suspect the port is not being correctly initialized

**Reply by hengweizhu** [reply]
Thanks Bruno that seemed to have made it work! I have another question: is it possible to set the strobe state of a GPIO from Bonsai? If so, how do I do it? And if currently not, is there a way to implement it?
